crt_portal/cts_forms/views.py

```diff
@@ -1462,6 +1467,14 @@ def post(self, request, id=None):
                 report.save()
                 report_dispo_object.rejected = True
                 report_dispo_object.save()
+            approved_report_dispo_objects = ReportDisposition.objects.filter(batch=batch).exclude(public_id__in=rejected_report_ids)
+            for report_dispo_object in approved_report_dispo_objects:
+                report = Report.objects.filter(public_id=report_dispo_object.public_id).first()
+                report.batched_for_disposal = True
+                report.report_disposition_status = 'approved'
+                report.save()
+                report_dispo_object.rejected = False
+                report_dispo_object.save()
```

Comment:
it's possible there's hundreds of these in a batch, right?

that could cause problems - especially if this fails partway through. It's important to remember that `filter` doesn't actually return a list of objects - it returns a queryset. The query is only run when data on the objects is accessed. Whenever you're iterating over a queryset, look for a "filter" + "bulk update" method, instead.

For example, something like this might take this (potentially hundreds of queries) down to two:

```py
approved_report_dispo_queryset = ReportDisposition.objects.filter(batch=batch).exclude(public_id__in=rejected_report_ids)
public_ids = approved_report_dispo_queryset.values_list('public_id', flat=True)

Report.objects.filter(public_id__in=public_ids).update(report_disposition_status='approved', batched_for_disposal=True)

approved_report_dispo_queryset.update(rejected=False)
```

[stackoverflow for efficient bulk update](https://stackoverflow.com/questions/41744096/efficient-way-to-update-multiple-fields-of-django-model-object)